dask_geopandas/hilbert_distance.py

```diff
@@ -0,0 +1,288 @@
+import numpy as np
+import pandas as pd
+from numba import jit
+
+ngjit = jit(nopython=True, nogil=True)
+
+# Based on: https://github.com/holoviz/spatialpandas/blob/
+# 9252a7aba5f8bc7a435fffa2c31018af8d92942c/spatialpandas/dask.py
+
+
+def _calculate_hilbert_distance(gdf, total_bounds, p):
+
+    """
+    Calculate hilbert distance for a GeoDataFrame
+    int coordinates
+
+    Parameters
+    ----------
+    gdf : GeoDataFrame
+
+    total_bounds : Total bounds of geometries - array
+
+    p : Hilbert curve parameter
+
+    Returns
+    ---------
+    Pandas Series containing hilbert distances
+    """
+
+    # Compute bounds as array
+    bounds = gdf.bounds.to_numpy()
+    # Compute hilbert distances
+    distances = _hilbert_distance(total_bounds, bounds, p)
+
+    return pd.Series(distances, name="hilbert_distance")
+
+
+@ngjit
+def _hilbert_distance(total_bounds, bounds, p):
+
+    """
+    Calculate the hilbert distance based on the total bounds and
+    bounds for a collection of geometries
+
+    Parameters
+    ----------
+
+    total_bounds : Total bounds of geometries - array
+
+    bounds : Bounds of each geometry - array
+
+    p : Hilbert curve parameter
+
+    Returns
+    ---------
+    Array of hilbert distances for each geom
+    """
+
+    coords = _continuous_to_discrete_coords(total_bounds, bounds, p)
+    # Calculate hilbert distance
+    distances = _distances_from_coordinates(p, coords)
+
+    return distances
+
+
+@ngjit
+def _continuous_to_discrete_coords(total_bounds, bounds, p):
+
+    """
+    Calculates mid points & ranges of geoms and returns
+    as discrete coords
+
+    Parameters
+    ----------
+
+    total_bounds : Total bounds of geometries - array
+
+    bounds : Bounds of each geometry - array
+
+    p : Hilbert curve parameter
+
+    Returns
+    ---------
+    Discrete two-dimensional numpy array
+    Two-dimensional array Array of hilbert distances for each geom
+    """
+
+    # Hilbert Side len
+    side_length = 2 ** p
+
+    # Calculate x and y range of total bound coords - returns array
+    geom_ranges = [
+        (total_bounds[0], total_bounds[2]),
+        (total_bounds[1], total_bounds[3]),
+    ]
```

Comment:
```suggestion
    xmin, ymin, xmax, ymax = total_bounds
```

and then passing `(xmin, xmax)` to `_continuous_to_discrete` below might be a bit simpler / more explicit to read (without the need to create a temporary geom_ranges variable that contains both ranges)